contracts/strategy/BnbxYieldConverterStrategy.sol

```diff
@@ -0,0 +1,245 @@
+        address stakeManager
+    ) public initializer {
+        __BaseStrategy_init(destination, feeRecipient);
+
+        _bnbxToken = IERC20Upgradeable(bnbxToken);
+        _stakeManager = IStakeManager(stakeManager);
+        _vault = IMasterVault(masterVault);
+
+        _bnbxToken.approve(stakeManager, type(uint256).max);
+    }
+
+    /**
+     * Modifiers
+     */
+    modifier onlyVault() {
+        require(msg.sender == address(_vault), "!vault");
+        _;
+    }
+
+    /// @dev deposits the given amount of BNB into Stader stakeManager through bnbxRouter
+    function deposit() external payable onlyVault returns (uint256) {
+        uint256 amount = msg.value;
+        require(amount <= address(this).balance, "insufficient balance");
+
+        return _deposit(amount);
+    }
+
+    /// @dev deposits all the available BNB into Stader stakeManager through bnbxRouter
+    function depositAll() external payable onlyVault returns (uint256) {
+        return _deposit(address(this).balance);
+    }
+
+    /// @dev internal function to deposit the given amount of BNB into Stader stakeManager through bnbxRouter
+    /// @param amount amount of BNB to deposit
+    function _deposit(uint256 amount)
+        internal
+        whenDepositNotPaused
+        returns (uint256)
+    {
+        require(canDeposit(amount), "invalid amount");
+
+        _bnbDepositBalance += amount;
+        _bnbxHoldingBalance += _stakeManager.convertBnbToBnbX(amount);
+        _stakeManager.deposit{value: amount}();
+        return amount;
+    }
+
+    /// @dev withdraws the given amount of BNB from Stader's stakeManager
+    /// @param amount amount of BNB to withdraw
+    function withdraw(address recipient, uint256 amount)
+        external
+        onlyVault
+        returns (uint256)
+    {
+        return _withdraw(recipient, amount);
+    }
+
+    /// @dev withdraws everything from Stader's stakeManager
+    function panic() external onlyStrategist returns (uint256) {
+        (, , uint256 debt) = _vault.strategyParams(address(this));
+        return _withdraw(address(_vault), debt);
+    }
+
+    /// @dev internal function to withdraw the given amount of BNB from Stader's stakeManager
+    /// @param amount amount of BNB
+    /// @return value - returns the amount of BNB withdrawn from stader (currently 0 as no amount is withdrawn in this step)
+    function _withdraw(address recipient, uint256 amount)
+        internal
+        returns (uint256 value)
+    {
+        require(amount > 0, "invalid amount");
+
+        uint256 bnbxAmount = _stakeManager.convertBnbToBnbX(amount);
+        _bnbDepositBalance -= amount;
+        _bnbxHoldingBalance -= bnbxAmount;
+
+        _bnbxToUnstake += bnbxAmount;
+        _withdrawRequests[_nextWithdrawIdx++] = UserWithdrawRequest({
+            recipient: recipient,
+            amount: amount,
+            bnbxAmount: bnbxAmount,
+            triggerTime: block.timestamp
+        });
+
+        return 0;
```

Comment:
I think we will have to return amount here. checked masterVault's code.
They are using this to update `debt`